File: Exercicio8/exercicio8.hs
divisores :: (Integral a) => a -> [a]
divisores x = [y | y <- [1..x-1], (x `mod` y) == 0]

perfeito :: (Integral a) => a -> Bool
perfeito x = x == (sum (divisores x))

main = do
    putStr "Digite um numero pra saber se ele é perfeito: "
    num <- getLine
    print (perfeito (read num :: Int))

File: Exercicio8/exercicio8.lua
numero=io.read("*n")
soma=0
for divisor=1,numero-1 do
if numero%divisor==0 then
soma=soma+divisor
end
end
if numero==soma then
print(numero.." E PERFEITO")
else
print(numero.." NAO E PERFEITO")
end